### PR TITLE
Split xcconfig to pod_target_xcconfig and user_target_xcconfig

### PR DESCRIPTION
**Pull request by mrackwitz**
* [x] Depends on CocoaPods/Core#242

Makes a first attempt in fixing #3465.

### CODE REVIEW

File: lib/cocoapods/generator/xcconfig/aggregate_xcconfig.rb [hunk 1]
```diff
@@ -133,6 +134,70 @@ def generate_ld_runpath_search_paths
...
+        # Merges the +user_target_xcconfig+ for all pod targets into the
+        # #xcconfig and warns on conflicting definitions.
+        #
+        # @return [Hash{String, String}]
+        #
+        def merged_user_target_xcconfigs
+          settings = user_target_xcconfig_values_by_target_by_key
+          settings.each_with_object({}) do |(key, values_by_target), xcconfig|
+            uniq_values = values_by_target.values.uniq
+            values_are_bools = values_by_target.values.all? { |v| v =~ /(yes|no)/i }
```
Comment (segiddins) [inline lib/cocoapods/generator/xcconfig/aggregate_xcconfig.rb]: This should check uniq values.

**Comment by segiddins** [comment]
@mrackwitz can you leave some notes explaining what makes this PR a WIP?

**Comment by mrackwitz** [comment]
@segiddins: Well, I hoped to get more feedback in #3465 as this a breaking change, which will break the linting for a lot of podspecs through the deprecation warning.

**Comment by segiddins** [comment]
Even so, please explain what it will take to make this not a `WIP`, that makes reviewing a lot easier.

**Comment by mrackwitz** [comment]
@segiddins: Once the Core PR is done, it should be ready.

**Comment by segiddins** [comment]
Please add a test covering `user_target_xcconfig`?

**Comment by segiddins** [comment]
:fireworks:

**Pull request merged**